vfs/mem_fs.go

```diff
@@ -657,8 +625,9 @@ func (f *memNode) resetToSyncedState() {
 	}
 }
 
-// memFile is a reader or writer of a node's data, and implements File.
+// memFile is a reader or writer of a node's data. Implements File.
 type memFile struct {
+	name        string
```

Comment:
We might also need a follow-up to remove this field from `memFile`, and "query" it from `MemFS` instead, when needed. The file name can be changed concurrently by `Rename()`, so this name can be stale.

Reply:
Nah, we don't need to do that. The "canonical" implementation of this interface in Go packages [returns](https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/os/stat_unix.go;l=24) the cached file name that was used to open it.